File: c++_codes/e1_sum_fibonachi.cpp
/* enunciados
calcule la suma de los terminos de la serie de FIBONACCI
cuyos valores se encuentran entrre 100 y 10,000 
*/
#include <iostream>
using namespace std;

int main(){
    // declaramos variables
    int p1=0;
    int p2=1;
    int p3;
    int sum_e=0;
    bool go=false;
    cout<<"valores del la serie de finonacci entre 100 y 10,000"<<endl;
    while(go==false){
        // obtenemos los valores de la serie
        p3 = p1 + p2; // los valores de la serie de fibonacci se obtienen sumando los dos ultimos numeros de la serie 
        
        if((p3>=100) && (p3<=10000)){
            cout<<"--"<<p3<<endl;
            sum_e+=p3; // si se cumple la condicion se va realisando la suma de los valores 
        }

        // ordenamos los valores de las dos ultimas posiciones de la serie para poder obtener el siguente valor
        p1 = p2;
        p2 = p3;

        if (p3>10000){
            break;
        }
    
    }
    cout<<"el valor de la suma de los numeros de la serie de fibonacci que estan entre 100 y 10000 es de:"<<sum_e;
    return 0;
}

File: 6to_semestre/js_code/e1_sum_fibonachi.js

console.log("calcule la suma de los terminos de la serie de FIBONACCI cuyos valores se encuentran entrre 100 y 10,000")
var fibo =[];
var a=0;
var b=1;
var c;
while (true) {
    c=a+b;
    if (c>=100 && c<=10000) {
        fibo.push(c);
    }
    else if (c>100) {
        break;
    }
    a=b;
    b=c;
}


var suma = fibo.reduce(function(acum, num){
    return acum+num
})
console.log("respuesta : "+suma)
